osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/ConfigurationManagementServiceTest.java

```diff
@@ -0,0 +1,290 @@
+    private ConfigurationDto configurationDto;
+    @Mock
+    private DomainCoreMapper domainCoreMapper;
+    @Mock
+    private OrganisationDomainService organisationDomainService;
+    @Mock
+    private DeviceDomainService deviceDomainService;
+    @Mock
+    private CorrelationIds ids;
+    @Mock
+    private Configuration configuration;
+    @Mock
+    private OsgpCoreRequestMessageSender osgpCoreRequestMessageSender;
+    @Mock
+    private Ssld ssld;
+    @Mock
+    private Device device;
+    @Captor
+    private ArgumentCaptor<RequestMessage> requestMessageArgumentCaptor;
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptorOne;
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptorTwo;
+    @Captor
+    private ArgumentCaptor<Integer> integerArgumentCaptor;
+    @Captor
+    private ArgumentCaptor<Long> longArgumentCaptor;
+    @Captor
+    private ArgumentCaptor<ResponseMessage> responseMessageArgumentCaptor;
+    @InjectMocks
+    @Qualifier("organisationDomainService")
+    private ConfigurationManagementService configurationManagementService;
+    private final String organisationIdentification = "a";
+    private final String deviceIdentification = "b";
+    private final String correlationUid = "c";
+    private OsgpException exception;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    @BeforeEach
+    public void setUp() throws NoSuchFieldException,
+            SecurityException, IllegalArgumentException, IllegalAccessException {
+        // make ids
+        this.ids = new CorrelationIds(this.organisationIdentification, this.deviceIdentification, this.correlationUid);
+        this.configurationManagementService = new ConfigurationManagementService();
+
+        // do injection using reflection
+        // Inject Mocks doesn't inject these mocks, possibly because they are in the parent class of
+        // ConfigurationManagementService
+        this.injectionUsingReflection(AbstractService.class, "organisationDomainService",
+                this.configurationManagementService, this.organisationDomainService);
+        this.injectionUsingReflection(AbstractService.class, "deviceDomainService", this.configurationManagementService,
+                this.deviceDomainService);
+        this.injectionUsingReflection(AbstractService.class, "domainCoreMapper", this.configurationManagementService,
+                this.domainCoreMapper);
+        this.injectionUsingReflection(AbstractService.class, "osgpCoreRequestMessageSender",
+                this.configurationManagementService, this.osgpCoreRequestMessageSender);
+        this.injectionUsingReflection(AbstractService.class, "webServiceResponseMessageSender",
+                this.configurationManagementService, this.webServiceResponseMessageSender);
+        this.injectionUsingReflection(AbstractService.class, "ssldRepository", this.configurationManagementService,
+                this.ssldRepository);
```

Comment:
remove this unnecessary code

Reply:
This code doesn't seem to be unnecessary. Mockito's injection with the Mock annotation doesn't seem to work, possibly because the fields are actually in the parent class AbstractService rather than in ConfigurationManagementService. Simply removing the code results in nullpointer exceptions because the mock is never inserted.

Reply:
it works without this code block

Reply:
you really should change this